test/Engine/ProtoTest/LiveRunnerTests/MicroFeatureTests.cs

```diff
@@ -5930,6 +5930,69 @@ public void TestMAGN5477()
             AssertValue("k", 3);
         }
 
+        [Test]
+        public void CrashOnUndoDelFuncDef()
+        {
+            List<string> codes = new List<string>()
+            {
+                @"
+                    def foo(a, b)
+                    {
+	                    return [Imperative]
+	                    {
+		                    list = a..b;
+		                    x = [];
+		                    count = 0;
+		                    for(i in list)
+		                    {
+			                    x[count] = i;
+			                    count = count+1;
+		                    }
+		                    return x;
+	                    }
+                    };      
+                ",
+                @"
+                    x = false;
+                    l1 = foo(1, 10);
+                    l2 = foo(10, 20);
+                    [Imperative]
+                    {
+	                    if(x==true)
+	                    {
+		                    return l1;
+	                    }
+	                    else
+	                    {
+		                    return l2;
+	                    }
+                    };      
+                "
+            };
+
+            Guid guidFuncDef = Guid.NewGuid();
+            Guid guidOther = Guid.NewGuid();
+
+            List<Subtree> added = new List<Subtree>();
+            added.Add(TestFrameWork.CreateSubTreeFromCode(guidFuncDef, codes[0]));
+            added.Add(TestFrameWork.CreateSubTreeFromCode(guidOther, codes[1]));
+
+            var syncData = new GraphSyncData(null, added, null);
+            liveRunner.UpdateGraph(syncData);
+
+            List<Subtree> removed = new List<Subtree>();
+            removed.Add(TestFrameWork.CreateSubTreeFromCode(guidFuncDef, codes[0]));
+  
+            syncData = new GraphSyncData(removed, null, null);
+            liveRunner.UpdateGraph(syncData);
```

Comment:
Can you add an assertion here that an exception is not thrown on running this code?

Reply:
Just curious, does this test fail without your fixes?

Reply:
Yes it fails with exception thrown.